ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -508,32 +534,36 @@ data TraceEvent blk
     -- ^ An old or invalid on-disk snapshot was deleted
   deriving (Generic, Eq, Show)
 
+
+-- | Which point the replay started from
+newtype ReplayStart blk = ReplayStart (Point blk) deriving (Eq, Show)
+
+-- | Which point the replay is expected to end at
+newtype ReplayGoal blk = ReplayGoal (Point blk) deriving (Eq, Show)
+
 -- | Events traced while replaying blocks against the ledger to bring it up to
 -- date w.r.t. the tip of the ImmutableDB during initialisation. As this
 -- process takes a while, we trace events to inform higher layers of our
 -- progress.
---
--- The @replayTo@ parameter is meant to be filled in by a higher layer,
--- i.e., the ChainDB.
-data TraceReplayEvent blk replayTo
-  = ReplayFromGenesis replayTo
+data TraceReplayEvent blk
```

Comment:
We don't necessarily need to use the new goal and start newtypes in here; we could just use `Point`. But the newtypes are probably clearer, ultimately. Even though they'll require some (minor and probably more legible!) swizzling by downstream users.

Reply:
I would keep them for now. They don't do any harm and we are the ones that will update the downstream deps so we can deal with it :smile: